apps/prairielearn/src/models/course-instance-publishing-extensions.ts

```diff
@@ -0,0 +1,182 @@
+import {
+  execute,
+  loadSqlEquiv,
+  queryOptionalRow,
+  queryRow,
+  queryRows,
+  runInTransactionAsync,
+} from '@prairielearn/postgres';
+
+import {
+  type CourseInstancePublishingEnrollmentExtension,
+  CourseInstancePublishingEnrollmentExtensionSchema,
+  type CourseInstancePublishingExtension,
+  CourseInstancePublishingExtensionSchema,
+} from '../lib/db-types.js';
+
+import {
+  type CourseInstancePublishingExtensionWithUsers,
+  CourseInstancePublishingExtensionWithUsersSchema,
+} from './course-instance-publishing-extensions.types.js';
+
+const sql = loadSqlEquiv(import.meta.url);
+
+/**
+ * Finds all publishing extensions that apply to a specific enrollment.
+ */
+export async function selectPublishingExtensionsByEnrollmentId(
+  enrollment_id: string,
+): Promise<CourseInstancePublishingExtension[]> {
+  return await queryRows(
+    sql.select_publishing_extensions_by_enrollment_id,
+    { enrollment_id },
+    CourseInstancePublishingExtensionSchema,
+  );
+}
+
+/**
+ * Finds a publishing extension by name within a course instance.
+ */
+export async function selectPublishingExtensionByName({
+  name,
+  course_instance_id,
```

Comment:
Shouldn't this and other model functions be accepting a full `course_instance`, not an ID?

Reply:
Yes, fixed